frontend/mulighetsrommet-veileder-flate/src/hooks/useQueryWrapper.ts

```diff
@@ -0,0 +1,54 @@
+import { useQuery, UseQueryOptions, useSuspenseQuery } from "@tanstack/react-query";
+
+// Define the type of your wrapped useQuery
+interface UseQueryWrapperOptions<TQueryFnData, TError, TData>
+  extends Omit<UseQueryOptions<TQueryFnData, TError, TData>, "queryFn"> {
+  queryFn: () => Promise<{ data: TQueryFnData }>;
+}
+
+/* Siden hey-api sin fetch-client returnerer { data, error } i stedet for kun
+ * data har vi denne utilitien som unwrapper så returobjektet ser likt ut som
+ * useQuery.
+ *
+ * Synes egentlig det er litt rart at dette trengs når vi setter throwOnError
+ * i hey-api (som vi gjør), kunne ikke den bare returnert data da? Kan hende
+ * det er en config et sted.
+ */
+export function useQueryWrapper<TQueryFnData, TError, TData = TQueryFnData>(
```

Comment:
Kunne kanskje kalt den `useApiQuery` eller noe lignende, for å gjøre det litt tydligere hvorfor vi har en custom query-wrapper